### ISSUE TITLE
Bug: new no-missing-link-fragments fails on valid URL-encoded ones

### ISSUE THREAD

**Issue by adamlui** [eslint/markdown#430]
### Environment

ESLint version: v9.29.0
@eslint/markdown version: 6.4.0
Node version: v22.15.0
npm version: v11.3.0
Operating System: win32 10.0.19045

### Which language are you using?

commonmark

### What did you do?

<details>
<summary>Configuration</summary>

```js
import js from '@eslint/js'
import globals from 'globals'
import importPlugin from 'eslint-plugin-import'
import json from '@eslint/json'
import markdown from '@eslint/markdown'
import regexp from 'eslint-plugin-regexp'
import stylisticJS from '@stylistic/eslint-plugin'
import yml from 'eslint-plugin-yml'

export default [
    { ignores: ['**/sandbox/', '**/*.min.js', '**/package-lock.json'] },
    {
        files: ['**/*.{js,mjs}'],
        languageOptions: {
            ecmaVersion: 'latest', sourceType: 'script',
            globals: { ...globals.browser, ...globals.node }
        },
        plugins: { 'import': importPlugin, 'js-styles': stylisticJS, regexp },
        rules: {
            ...js.configs.recommended.rules,
            ...importPlugin.flatConfigs.recommended.rules,
            ...regexp.configs['flat/recommended'].rules,
            'indent': 'off', 'no-unexpected-multiline': 'off', 'key-spacing': 'off', // allow whitespace anywhere
            'js-styles/no-trailing-spaces': 'error', // ...except at ends of lines
            'js-styles/max-len': ['error', { 'code': 120, // limit lines to 120 chars except if containing...
                'ignoreComments': true, 'ignoreStrings': true, // ...trailing/own-line comments, quoted strings...
                'ignoreTemplateLiterals': true, 'ignoreRegExpLiterals': true }], // ...or template/regex literals
            'js-styles/no-extra-semi': 'error', // disallow unnecessary semicolons
            'quotes': ['error', 'single', // enforce single quotes...
                { 'allowTemplateLiterals': true }], // ...except backticks to avoid escaping quotes
            'comma-dangle': ['error', 'never'], // enforce no trailing commas in arrays or objects
            'no-constant-condition': 'off', // allow constant conditions
            'no-empty': 'off', // allow empty blocks
            'no-inner-declarations': 'off', // allow function declarations anywhere
            'no-useless-escape': 'off', // allow all escape chars cause ESLint sucks at detecting truly useless ones
            'no-unused-vars': ['error', { 'caughtErrors': 'none' }], // allow unused named args in catch blocks
            'import/no-unresolved': ['error', { ignore: ['^(?:https?://|imagemin)'] }] // enforce import res for ESM
                // + ignore false positives ( https://github.com/import-js/eslint-plugin-import/issues/3118
                //                          + https://github.com/import-js/eslint-plugin-import/issues/3119 )
        }
    },
    { files: ['**/*.json'], language: 'json/json', ...json.configs.recommended },
    {
        files: ['**/*.md'], language: 'markdown/commonmark', plugins: { markdown },
        rules: {
            ...markdown.configs.recommended[0].rules,
            'markdown/heading-increment': 'off', // allow headings to skip levels
            'markdown/fenced-code-language': 'off', // allow code blocks w/ no language specified
            'markdown/no-missing-label-refs': 'off' // allow missing label references
        }
    },
    { files: ['**/*.mjs'], languageOptions: { sourceType: 'module' }},
    { files: ['**/*.{yaml,yml}'], ...yml.configs['flat/standard'][1] }
]
```
</details>

```md
...
যখন ইনস্টল করা হয় [বিশ্বব্যাপী](#-%E0%A6%B8%E0%A7%8D%E0%A6%A5%E0%A6%BE%E0%A6%AA%E0%A6%A8), **generate-pw** কমান্ড লাইন থেকেও ব্যবহার করা যেতে পারে। মৌলিক কমান্ড হল:
```

### What did you expect to happen?

No error

### What actually happened?

![Image](https://github.com/user-attachments/assets/be1eac92-70b0-416b-9844-1a3434794519)

### Link to Minimal Reproducible Example

https://github.com/adamlui/js-utils/tree/main/generate-pw/docs/bn#-%E0%A6%95%E0%A6%AE%E0%A6%BE%E0%A6%A8%E0%A7%8D%E0%A6%A1-%E0%A6%B2%E0%A6%BE%E0%A6%87%E0%A6%A8-%E0%A6%AC%E0%A7%8D%E0%A6%AF%E0%A6%AC%E0%A6%B9%E0%A6%BE%E0%A6%B0 (click 1st blue link the fragment is URL-encoded and works)

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by snitin315** [comment]
I was able to reproduce as well, sharing a minimal repro:

```md
<!-- eslint markdown/no-missing-link-fragments: "error" -->

# test

দেখা: [উপলব্ধ বিকল্প](#generate-%E0%A6%AB%E0%A6%BE%E0%A6%82%E0%A6%B6%E0%A6%A8%E0%A7%87%E0%A6%B0-%E0%A6%9C%E0%A6%A8%E0%A7%8D%E0%A6%AF-%E0%A6%89%E0%A6%AA%E0%A6%B2%E0%A6%AC%E0%A7%8D%E0%A6%A7-%E0%A6%AC%E0%A6%BF%E0%A6%95%E0%A6%B2%E0%A7%8D%E0%A6%AA)

## `generate*()` ফাংশনের জন্য উপলব্ধ বিকল্প

প্রতিটি `generate*()` ফাংশনের জন্য বিকল্প অবজেক্টে এর যেকোনো একটি পাস করা যেতে পারে:
```

It is also not reported by [MD051](https://github.com/DavidAnson/markdownlint/blob/main/doc/md051.md)

@xbinaryx What fix do you have in mind for this?
